otj-metrics-core/src/main/java/com/opentable/metrics/ready/ReadyCheckRegistry.java

```diff
@@ -0,0 +1,243 @@
+     *
+     * @param name the name of the {@link ReadyCheck} instance
+     */
+    public void unregister(String name) {
+        ReadyCheck readyCheck;
+        synchronized (lock) {
+            readyCheck = readyChecks.remove(name);
+        }
+        if (readyCheck != null) {
+            onReadyCheckRemoved(name, readyCheck);
+        }
+    }
+
+    /**
+     * Returns a set of the names of all registered ready checks.
+     *
+     * @return the names of all registered ready checks
+     */
+    public SortedSet<String> getNames() {
+        return Collections.unmodifiableSortedSet(new TreeSet<>(readyChecks.keySet()));
+    }
+
+    /**
+     * Returns the {@link ReadyCheck} instance with a given name
+     *
+     * @param name the name of the {@link ReadyCheck} instance
+     */
+    public ReadyCheck getReadyCheck(String name) {
+        return readyChecks.get(name);
+    }
+
+    /**
+     * Runs the ready check with the given name.
+     *
+     * @param name the ready check's name
+     * @return the result of the ready check
+     * @throws NoSuchElementException if there is no ready check with the given name
+     */
+    public ReadyCheck.Result runReadyCheck(String name) throws NoSuchElementException {
+        final ReadyCheck readyCheck = readyChecks.get(name);
+        if (readyCheck == null) {
+            throw new NoSuchElementException("No ready check named " + name + " exists");
+        }
+        return readyCheck.execute();
+    }
+
+    /**
+     * Runs the registered ready checks matching the filter and returns a map of the results.
+     *
+     * @return a map of the ready check results
+     */
+    public SortedMap<String, ReadyCheck.Result> runReadyChecks() {
+        final SortedMap<String, ReadyCheck.Result> results = new TreeMap<>();
+        for (Map.Entry<String, ReadyCheck> entry : readyChecks.entrySet()) {
+            final ReadyCheck.Result result = entry.getValue().execute();
+            results.put(entry.getKey(), result);
+        }
+        return Collections.unmodifiableSortedMap(results);
+    }
+
+
+    /**
+     * Runs the registered ready checks matching the filter in parallel and returns a map of the results.
+     *
+     * @param executor object to launch and track ready checks progress
+     * @return a map of the ready check results
+     */
+    public SortedMap<String, ReadyCheck.Result> runReadyChecks(ExecutorService executor) {
+        final Map<String, Future<ReadyCheck.Result>> futures = new HashMap<>();
+        for (final Map.Entry<String, ReadyCheck> entry : readyChecks.entrySet()) {
+            final String name = entry.getKey();
+            final ReadyCheck readyCheck = entry.getValue();
+            futures.put(name, executor.submit(readyCheck::execute));
+        }
+
+        final SortedMap<String, ReadyCheck.Result> results = new TreeMap<>();
+        for (Map.Entry<String, Future<ReadyCheck.Result>> entry : futures.entrySet()) {
```

Comment:
It does mean this yes. I've added an injectable timeout. Since we need ALL results, there's not a lot of point in complete parallization IMO, since ready and health checks should be light weight.

Reply:
Actually I removed the timeout, because it was a PITA to inject.